operator/v1/techpreview.machineconfiguration.testsuite.yaml

```
@@ -115,3 +115,212 @@ tests:
+            - path: "/etc/my-file"
+              actions:
+                - type: DaemonReload
+                - type: Reload
+                  reload:
+                    serviceName: my.service
+          units:
+            - name: "my.service"
+              actions:
+                - type: Restart
+                  restart:
+                    serviceName: my-other.service
+          sshkey:
+            actions:
+              - type: None
+  - name: Should be able to define a node disruption policy with multiple entries of the same type
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: MachineConfiguration
+      spec:
+        nodeDisruptionPolicy:
+          files:
+            - path: "/etc/my-file"
+              actions:
+                - type: DaemonReload
+                - type: Reload
+                  reload:
+                    serviceName: my.service
+            - path: "/etc/my-other-file"
+              actions:
+                - type: Drain
+                - type: Restart
+                  restart:
+                    serviceName: my.service
+    expected: |
+      apiVersion: operator.openshift.io/v1
+      kind: MachineConfiguration
+      spec:
+        logLevel: Normal
+        operatorLogLevel: Normal
+        nodeDisruptionPolicy:
+          files:
+            - path: "/etc/my-file"
+              actions:
+                - type: DaemonReload
+                - type: Reload
+                  reload:
+                    serviceName: my.service
+            - path: "/etc/my-other-file"
+              actions:
+                - type: Drain
+                - type: Restart
+                  restart:
+                    serviceName: my.service
+  - name: Node disruption policies with duplicate file entries should fail
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: MachineConfiguration
+      spec:
+        nodeDisruptionPolicy:
+          files:
+            - path: "/etc/my-file"
+              actions:
+                - type: DaemonReload
+                - type: Reload
+                  reload:
+                    serviceName: my.service
+            - path: "/etc/my-file"
+              actions:
+                - type: Drain
+                - type: Reload
+                  reload:
+                    serviceName: my.service
+    expectedError: "spec.nodeDisruptionPolicy.files[1]: Duplicate value: map[string]interface {}{\"path\":\"/etc/my-file\"}"
+  - name: Node disruption policies that specify a reload action should fail if you don't specify the reload field
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: MachineConfiguration
+      spec:
+        nodeDisruptionPolicy:
+          files:
+            - path: "/etc/my-file"
+              actions:
+                - type: DaemonReload
+                - type: Reload
+    expectedError: "Reload is required when type is reload, and forbidden otherwise"
```

This is the wrong way around, think about it from an end user perspective, they see the field as lower r and the type as upper r
```suggestion
    expectedError: "reload is required when type is Reload, and forbidden otherwise"
```